queries/vala/highlights.scm

```diff
@@ -0,0 +1,288 @@
+; Variable
+
+(uppercased_identifier) @constant
+
+(namespaced_identifier
+  left: [
+    (camel_cased_identifier) @namespace
+    (identifier) @variable
+  ]
+  right: [
+    (identifier) @parameter
+    (camel_cased_identifier) @type
+    (uppercased_identifier) @constant
+  ]
+)
+; Pointers
+
+(address_of_identifier "&" @symbol)
+(pointer_type "*" @symbol)
+(indirection_identifier "*" @symbol)
+
+; Misc
+
+(number) @number
+
+[
+  "{"
+  "}"
+  "("
+  ")"
+] @punctuation.bracket
```

Comment:
I think `"["` `"]"` is missing. Could you add those two details please? Just tried out your vala highlighting for the first time.

Reply:
`void open (File[] files, string hint) {` array_identifier also does not get highlighted as a type. But that can be a follow up

Reply:
My bad, must have removed them somewhere during simplification

Reply:
> `void open (File[] files, string hint) {` array_identifier also does not get highlighted as a type. But that can be a follow up

![image](https://user-images.githubusercontent.com/48367298/151671127-1fe622db-d490-4ebb-bd4e-3b01216327ef.png)

I believe it was, my last commit seems to fix it.